### PR TITLE
bpo-31234: threading_cleanup() now warns immediately

### PR DESCRIPTION
**Pull request by vstinner**
support.threading_cleanup() waits for 1 second before emitting a
warning if there are threads running in the background. With this
change, it now emits the warning immediately, to be able to catch
bugs more easily.

https://bugs.python.org/issue31234

### CODE REVIEW

File: Lib/test/support/__init__.py [hunk 1]
```diff
@@ -2039,22 +2039,23 @@ def threading_cleanup(*original_values):
...
-    t0 = time.monotonic()
+
     for count in range(_MAX_COUNT):
         values = _thread._count(), threading._dangling
         if values == original_values:
             break
+
+        if not count:
+            environment_altered = True
+            print("Warning -- threading_cleanup() detected %s leaked threads "
```
Comment (pitrou) [inline Lib/test/support/__init__.py]: I don't like the "leak" terminology here. I think the previous message was better.

**Comment by vstinner** [comment]
I rebased my change to get other fixes for dangling threads and processes.

File: Lib/test/support/__init__.py (hunk 1 above)
Comment (vstinner) [reply Lib/test/support/__init__.py]: Fixed. I reverted to the original message.

**Comment by vstinner** [comment]
test_ftplib and test_poplib should have been fixed by PR #3540. I rebased my PR one more time, only test_ssl should fail. Sadly, I'm unable to reproduce the test_ssl failure.

**Comment by vstinner** [comment]
I tag again the PR as [WIP] until the test_ssl issue is fixed.

I rebased the PR to get the second test_ftplib fix (PR #3544).

**Pull request merged**